### PR TITLE
FSE REST permissions endpoint.

### PR DESCRIPTION
**Pull request by peterwilsoncc**
Unlinked to the ticket while I work a few things out: mainly that I am not about to break everything.

cc @spacedmonkey @TimothyBJacobs @costdev

### CODE REVIEW

File: src/wp-includes/rest-api/endpoints/class-wp-rest-global-styles-controller.php [hunk 1]
```diff
@@ -104,6 +104,7 @@ protected function permissions_check() {
 	 * @return true|WP_Error True if the request has read access for the item, WP_Error object otherwise.
 	 */
 	public function get_item_permissions_check( $request ) {
+		return current_user_can( 'read_post', $request['id'] );
```
Comment (costdev) [inline src/wp-includes/rest-api/endpoints/class-wp-rest-global-styles-controller.php:107]: This returns `false` when the user does not have read access, but this method should return `true|WP_Error` for consistency within the REST API.

[See here](https://github.com/WordPress/wordpress-develop/blob/7b9e33a1e76c5052b57f7a701e5ec228b3751dea/src/wp-includes/rest-api/endpoints/class-wp-rest-menus-controller.php#L101) for an example `WP_Error`.

File: src/wp-includes/rest-api/endpoints/class-wp-rest-global-styles-controller.php [hunk 2]
```diff
@@ -134,6 +135,7 @@ public function get_item( $request ) {
 	 * @return true|WP_Error True if the request has write access for the item, WP_Error object otherwise.
 	 */
 	public function update_item_permissions_check( $request ) {
+		return current_user_can( 'edit_post', $request['id'] );
```
Comment (costdev) [inline src/wp-includes/rest-api/endpoints/class-wp-rest-global-styles-controller.php:138]: This returns `false` when the user does not have write access, but this method should return `true|WP_Error` for consistency within the REST API.

[See here](https://github.com/WordPress/wordpress-develop/blob/7b9e33a1e76c5052b57f7a701e5ec228b3751dea/src/wp-includes/rest-api/endpoints/class-wp-rest-menus-controller.php#L101) for an example `WP_Error`.

**Review by costdev — CHANGES_REQUESTED** [review]
Just a couple of quick notes on return values vs documentation.

**Comment by peterwilsoncc** [comment]
Thanks @costdev I'll close this down in favor of Jonny's in #1955 as it seems much closer to completion.

**Pull request closed without merging**